src/main/java/uk/gov/companieshouse/api/accounts/controller/DirectorsController.java

```diff
@@ -0,0 +1,145 @@
+import uk.gov.companieshouse.api.accounts.utility.LoggingHelper;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+
+import javax.servlet.http.HttpServletRequest;
+
+@RestController
+@RequestMapping("/transactions/{transaction_id}/company-accounts/{company_accounts_id}/small-full/directors-report/directors")
+public class DirectorsController {
+
+    @Autowired
+    private DirectorService directorService;
+
+    @Autowired
+    private ApiResponseMapper apiResponseMapper;
+
+    @Autowired
+    private ErrorMapper errorMapper;
+
+
+    @PostMapping
+    public ResponseEntity create(@RequestBody Director director, BindingResult bindingResult,
+                                 @PathVariable("companyAccountId") String companyAccountId, HttpServletRequest request) {
+
+        if (bindingResult.hasErrors()) {
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<Director> response =
+                    directorService.create(director, transaction, companyAccountId, request);
+
+            return apiResponseMapper
+                    .map(response.getStatus(), response.getData(), response.getErrors());
+
+        } catch (DataException ex) {
+
+            LoggingHelper.logException(companyAccountId, transaction,
+                    "Failed to create a Director resource", ex, request);
+            return apiResponseMapper.getErrorResponse();
+        }
+    }
+
+    @GetMapping
+    public ResponseEntity get(@PathVariable("companyAccountId") String companyAccountId,
+                              HttpServletRequest request) {
+
+        Transaction transaction = (Transaction) request
+                .getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<Director> response = directorService
+                    .find(companyAccountId, request);
+
+            return apiResponseMapper.mapGetResponse(response.getData(), request);
+
+        } catch (DataException ex) {
+
+            LoggingHelper.logException(companyAccountId, transaction,
+                    "Failed to retrieve a Director resource", ex, request);
+            return apiResponseMapper.getErrorResponse();
+        }
+    }
+
+    @PutMapping
+    public ResponseEntity update(@RequestBody Director director, BindingResult bindingResult,
+                                 @PathVariable("companyAccountId") String companyAccountId,
+                                 HttpServletRequest request) {
+
+        DirectorsReport directorsReport = (DirectorsReport) request.getAttribute(AttributeName.DIRECTORS_REPORT.getValue());
+        if (directorsReport.getLinks().get(SmallFullLinkType.DIRECTOR.getLink()) == null) {
```

Comment:
The logic isn't quite right here; we're updating a particular director, so we should check that `directorsReport.getDirectors().get(directorId)` is not null, where `directorId` is the id of the director undergoing change - this id is accessible as a PathVariable.